disco_aws_automation/disco_aws.py

```diff
@@ -394,6 +394,24 @@ def provision(self, ami, hostclass=None, owner=None, instance_type=None, monitor
 
         elb = self.update_elb(hostclass, update_autoscaling=False, testing=testing)
 
+        tags = {"hostclass": hostclass,
+                "owner": user_data["owner"],
+                "environment": self.environment_name,
+                "chaos": chaos,
+                "is_testing": "1" if testing else "0"}
+
+        if self.vpc.environment_class:
+            tags['environment_class'] = self.vpc.environment_class
+
+        hostclass_option_tags = self.hostclass_option_default(hostclass, "tags")
+        if hostclass_option_tags:
+            hostclass_tags = {k.strip(): v.strip() for k, v in
+                              [tag.split(':') for tag in hostclass_option_tags.split(',')]}
+        else:
```

Comment:
As it stands, this else doesn't seem very necessary. You could just do the update after formulating the `hostclass_tags`.